test/odbc/test_pg_datatype_decimal.cpp

```diff
@@ -0,0 +1,613 @@
+#include <gtest/gtest.h>
+#include <sqlext.h>
+#include "odbc_handler.h"
+#include "query_generator.h"
+
+using std::pair;
+
+const string TABLE_NAME = "master_dbo.decimal_table_odbc_test";
+const string VIEW_NAME = "master_dbo.decimal_view_odbc_test";
+const string DATATYPE = "sys.decimal";
+const int NUM_COLS = 5;
+const string COL_NAMES[NUM_COLS] = {"pk_1_0", "dec_5_5", "dec_38_38", "dec_38_0", "dec_4_2"};
+const int COL_PRECISION[NUM_COLS] = {1, 5, 38, 38, 4};
+const int COL_SCALE[NUM_COLS] = {0, 5, 38, 0, 2};
+
+// unsure of how these values are calculated, but they match with postgres' (non-sys) decimal type
+// Postgres documentation just says they're bytes size are variable
+const int LENGTH_EXPECTED[NUM_COLS] = {3, 5, 38, 38, 4}; 
+
+const string COL_TYPES[NUM_COLS] = {
+  DATATYPE + "(" + std::to_string(COL_PRECISION[0]) + "," + std::to_string(COL_SCALE[0]) +  ")",
+  DATATYPE + "(" + std::to_string(COL_PRECISION[1]) + "," + std::to_string(COL_SCALE[1]) + ")",
+  DATATYPE + "(" + std::to_string(COL_PRECISION[2]) + "," + std::to_string(COL_SCALE[2]) + ")",
+  DATATYPE + "(" + std::to_string(COL_PRECISION[3]) + "," + std::to_string(COL_SCALE[3]) + ")",
+  DATATYPE + "(" + std::to_string(COL_PRECISION[4]) + "," + std::to_string(COL_SCALE[4]) + ")"
+};
+
+vector<pair<string, string>> TABLE_COLUMNS = {
+  {COL_NAMES[0], COL_TYPES[0] + " PRIMARY KEY"},
+  {COL_NAMES[1], COL_TYPES[1]},
+  {COL_NAMES[2], COL_TYPES[2]},
+  {COL_NAMES[3], COL_TYPES[3]},
+  {COL_NAMES[4], COL_TYPES[4]}
+};
+
+const string MAX_DEC_5_5 = "0.99999";
+const string MAX_DEC_38_38 = "0.99999999999999999999999999999999999999";
+const string MAX_DEC_38_0 = "99999999999999999999999999999999999999";
+const string MAX_DEC_4_2 = "99.99";
+
+class PSQL_DataTypes_Decimal : public testing::Test{
+
+  void SetUp() override {
+
+    OdbcHandler test_setup;
+    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+
+  void TearDown() override {
+
+    OdbcHandler test_cleanup;
+    test_cleanup.ConnectAndExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+    test_cleanup.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+};
+
+// pads 0 at the end of the results depending on the scale number
+string FormatDecWithScale(string decimal, const int &scale) {
```

Comment:
I need to create this function because we always retrieves the values  with the same amount of decimals based on the scale. So if I insert 15, and it has scale of 3, I would get 15.000. This function just modifies the expected value to match it.